src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/SQLFunctions.java

```diff
@@ -710,6 +750,96 @@ private String upper(String property, String culture) {
         return new Tuple<>(name, def(name, "(int) " + getPropertyOrStringValue(field) + ".charAt(0)"));
     }
 
+    private Tuple<String, String> ifFunc(List<KVValue> paramers) {
+        String expr1 = paramers.get(1).value.toString();
+        String expr2 = paramers.get(2).value.toString();
+
+        /** Input with null is regarded as false */
+        if (paramers.get(0).value instanceof SQLNullExpr) {
+            return ifFunc(false, expr1, expr2);
+        }
+        if (paramers.get(0).value instanceof MethodField) {
+            String condition = getScriptText((MethodField) paramers.get(0).value);
+            return ifFunc(condition, expr1, expr2);
+        } else if (paramers.get(0).value instanceof SQLBooleanExpr) {
+            Boolean condition = ((SQLBooleanExpr) paramers.get(0).value).getValue();
+            return ifFunc(condition, expr1, expr2);
+        } else {
+            /**
+             * Detailed explanation of cases that come here:
+             * the condition expression would be in the format of a=b:
+             * a is parsed as the key (String) of a KVValue (get from paramers.get(0))
+             * and b is parsed as the value (Object) of this KVValue.
+             *
+             * Either a or b could be a column name, literal, or a number:
+             * - if isNumeric is true --> number
+             * - else if this string is single quoted --> literal
+             * - else --> column name
+             */
+            String key = getPropertyOrValue(paramers.get(0).key);
+            String value = getPropertyOrValue(paramers.get(0).value.toString());
+            String condition = key + " == " + value;
+            return ifFunc(condition, expr1, expr2);
+        }
+    }
+
+    private Tuple<String, String> ifFunc(String condition, String expr1, String expr2) {
+        String name = nextId("if");
+        return new Tuple<>(name, "boolean cond = " + condition + ";"
+                + def(name, "cond ? " + expr1 + " : " + expr2));
+    }
+
+    private Tuple<String, String> ifFunc(Boolean condition, String expr1, String expr2) {
+        String name = nextId("if");
+        if (condition) {
+            return new Tuple<>(name, def(name, expr1));
+        } else {
+            return new Tuple<>(name, def(name, expr2));
+        }
+    }
```

Comment:
Maybe we can inline these 2 functions into `ifFunc(List<KVValue> paramers)` just like your `ifnull` method because:

 1. They're short
 2. They share same logic `String name = nextId("if");`
 3. A little hard to follow 3 overload methods of same name.

Reply:
ok will do.

Reply:
Thanks.